nautobot_secrets_providers/tests/test_providers.py

```diff
@@ -202,3 +248,143 @@ def test_retrieve_invalid_key(self, requests_mocker):
 
         exc = err.exception
         self.assertIn(self.secret.parameters["key"], exc.message)
+
+    @requests_mock.Mocker()
+    @patch("builtins.open", new_callable=mock_open, read_data="data")
+    def test_get_client_k8s(self, requests_mocker, mock_file):
+        """Test Kubernetes Authentication."""
+        vault_url = "http://localhost:8200"
+        k8s_token_path = "/some/file/path"  # nosec B105
+        new_plugins_config = {
+            "nautobot_secrets_providers": {
+                "hashicorp_vault": {
+                    "url": vault_url,
+                    "auth_method": "kubernetes",
+                    "k8s_token_path": k8s_token_path,
+                },
+            },
+        }
+
+        # Test without specifying a role_name
+        with self.settings(PLUGINS_CONFIG=new_plugins_config):
+            with self.assertRaises(exceptions.SecretProviderError):
+                self.provider.get_client(self.secret)
```

Comment:
Might be good to catch the exception and assert that it has the correct/expected text in each such case?

```suggestion
            with self.assertRaises(exceptions.SecretProviderError) as context:
                self.provider.get_client(self.secret)
            self.assertEqual(str(context.exception), "...")
```